pkg/rancher-desktop/main/update/LonghornProvider.ts

```diff
@@ -186,6 +206,79 @@ export async function setHasQueuedUpdate(isQueued: boolean): Promise<void> {
   }
 }
 
+async function getPlatformVersion(): Promise<string> {
+  switch (process.platform) {
+  case 'win32':
+    return os.release();
+  case 'darwin': {
+    const macOsVersion = await getMacOsVersion(console);
+
+    if (macOsVersion === null) {
+      throw new Error('failed to get macOS version');
+    }
+
+    return macOsVersion.toString();
+  }
+  case 'linux':
+    // OS version is hard to get on Linux. Luckily, automatic updates
+    // are not supported on Linux as of the time of writing. "unknown"
+    // should suffice.
+    return 'unknown';
+  }
+  throw new Error(`Platform "${ process.platform }" is not supported`);
+}
+
+/**
+ * Fetch info on available versions of Rancher Desktop, as well as other
+ * things, from the Upgrade Responder server.
+ */
+export async function queryUpgradeResponder(url: string, currentVersion: semver.SemVer): Promise<UpgradeResponderQueryResult> {
+  const requestPayload = {
+    appVersion: currentVersion,
+    extraInfo:  {
+      platform:        `${ process.platform }-${ os.arch() }`,
+      platformVersion: await getPlatformVersion(),
+    },
+  };
+  // If we are using anything on `github.io` as the update server, we're
+  // trying to run a simplified test.  In that case, break the protocol and do
+  // a HTTP GET instead of the HTTP POST with data we should do for actual
+  // longhorn upgrade-responder servers.
+  const requestOptions = /^https?:\/\/[^/]+\.github\.io\//.test(url) ? { method: 'GET' } : {
+    method: 'POST',
+    body:   JSON.stringify(requestPayload),
+  };
+
+  console.debug(`Checking for upgrades from ${ url }`);
+  const responseRaw = await fetch(url, requestOptions);
+  const response = await responseRaw.json() as LonghornUpgraderResponse;
+
+  console.debug(`Upgrade server response:`, util.inspect(response, true, null));
+
+  // If the upgrade responder does not send the Supported field,
+  // assume that the version is supported.
+  for (const version of response.versions) {
+    version.Supported ??= true;
+  }
+
+  const allVersions = response.versions;
```

Comment:
If you create a temporary variable, why not create it earlier so you can use it in the code above?

```suggestion
  const allVersions = response.versions;

    for (const version of allVersions) {
    version.Supported ??= true;
  }
```